### PR TITLE
new resource `azurerm_stream_analytics_output_synapse`

### PR DESCRIPTION
**Pull request by owenfarrell**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### Summary

This pull request adds a new resource to manage a Stream Analytics output to a Synapse Workspace.

Fixes: #8881

Output from acceptance testing:

```
$ make acctests SERVICE='streamanalytics' TESTARGS='-run="^TestAccStreamAnalyticsOutputSynapse_"'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./internal/services/streamanalytics -run="^TestAccStreamAnalyticsOutputSynapse_" -timeout 180m -ldflags="-X=github.com/hashicorp/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccStreamAnalyticsOutputSynapse_basic
=== PAUSE TestAccStreamAnalyticsOutputSynapse_basic
=== RUN   TestAccStreamAnalyticsOutputSynapse_sqlPool
=== PAUSE TestAccStreamAnalyticsOutputSynapse_sqlPool
=== RUN   TestAccStreamAnalyticsOutputSynapse_update
=== PAUSE TestAccStreamAnalyticsOutputSynapse_update
=== RUN   TestAccStreamAnalyticsOutputSynapse_requiresImport
=== PAUSE TestAccStreamAnalyticsOutputSynapse_requiresImport
=== CONT  TestAccStreamAnalyticsOutputSynapse_basic
=== CONT  TestAccStreamAnalyticsOutputSynapse_update
=== CONT  TestAccStreamAnalyticsOutputSynapse_requiresImport
=== CONT  TestAccStreamAnalyticsOutputSynapse_sqlPool
--- PASS: TestAccStreamAnalyticsOutputSynapse_basic (629.72s)
--- PASS: TestAccStreamAnalyticsOutputSynapse_requiresImport (794.48s)
--- PASS: TestAccStreamAnalyticsOutputSynapse_update (895.11s)
--- PASS: TestAccStreamAnalyticsOutputSynapse_sqlPool (1147.35s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/streamanalytics       1150.196s
```

### CODE REVIEW

File: website/docs/r/stream_analytics_output_synapse.html.markdown [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+* `password` - (Required) The password that will be used to connect to the Azure SQL database. Changing this forces a new resource to be created.
+
+* `table` - (Required) The name of the table in the Azure SQL database. Changing this forces a new resource to be created.
+
+## Attributes Reference
+
+The following attributes are exported in addition to the arguments listed above:
+
+* `id` - The ID of the Stream Analytics Output Microsoft SQL Server Database.
```
Comment (stephybun) [inline website/docs/r/stream_analytics_output_synapse.html.markdown:86]: Wondering whether this should actually be
```suggestion
* `id` - The ID of the Stream Analytics Output Azure Synapse Workspace.
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @owenfarrell. This is almost good to go, just a few suggestions and waiting for the tests to run through 🙂.

**Comment by owenfarrell** [comment]
@stephybun - All updates incorporated. I may have gone overboard on the stream analytics type calculation, but I tried to replicate the existing pattern, which should help incorporate type validation in to the other stream analytics output resource types as well.

**Review by katbyte — APPROVED** [review]
Thanks @owenfarrell! LGTM 🚀

**Pull request merged**